### DISCUSSION TITLE
Azure AI search

### DISCUSSION THREAD

**Discussion by tgoshev** [microsoft/sample-app-aoai-chatGPT#721] [Q&A]
Hello, I have successfuly connected tha app to existing Azure resources (including Azure AI Search) and it is working fine.
Now, when debugging I could not find where the request to Azure AI Search is made and also how the app gets "intent" for the user question.

Could you please show me the method in which http request to Azure AI Search service is sent and also explain how the app gets the user intent?

**Comment by cbutler-captech** [comment]
It is in app.py.  The code below determines if you have data configured.  Then sends a bit of extra json to the chat completion to tell it what data to use.  In this case, it would be what index to use to search.  The intent is done by AOAI and uses the data to answer the question.

    if SHOULD_USE_DATA:
        model_args["extra_body"] = {
            "dataSources": [get_configured_data_source()]
        }

**Comment by tgoshev** [comment]
Yes, thank you. I was suspecting that everything is sent in one request to Azure Open AI API. 
**So, is this means that:**
   1. Azure Open AI internally gets user intent from the conversation so far (using "messages"), 
   2. generates relevant questions based on the intent and queries the Datasource (in this case Azure AI Search)
   3. and return appropriate Open AI generated answer based on the retrieved content.
  
**If this is the case do we have any control on how the datasource is queried and which content retrieved from the datasource is considered relevant to the question?**
   
I am showing an example of a chat about instructions for using shredder in the office. You can notice that for the second question it correctly gets the context from previous messages and sends relevant questions to the Datasource. 
I suppose that it sends all 3 questions to Datasource (What to do if a shredder is broken?, Steps to take if a shredder is broken, How to handle a broken shredder) and then combines the retrieved content, rerank it and use the best one as a source for the answer.

**Q1: "how to use a shredder"**

Message1:{"content": "**how to use a shredder**","role": "user"}
Message2:{"content": "{\"citations\": [{\"content\":"...\", \"filepath\": "...", \"chunk_id\": "..."}], \"**intent**\": \"[\\\"**How to use a shredder**\\\", \\\"**Instructions for using a shredder**\\\", \\\"**Steps to operate a shredder**\\\"]\"}","role": "tool"}
Message3:{'role': 'assistant', 'content': "The document provides instructions for using a shredder ..."}

A1: "The document provides instructions for using a shredder. Here are the key points:..."

**Q2: "what to do if it is broken"**

Message4:{'role': 'user', 'content': '**what to do if it is broken**'}
Message5:{'role': 'tool', 'content': '{"citations": [{"content": "...", "filepath":...", "chunk_id": "..."}], "**intent**": "[\\"**What to do if a shredder is broken?**\\", \\"**Steps to take if a shredder is broken**\\", \\"**How to handle a broken shredder**\\"]"}'}
Message6:{'role': 'assistant', 'content': 'The document provides specific actions to take if the shredder is broken....'}
